config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  resources :dishes do
+    resources :ingredients
+    resources :dish_ingredients, only: [:create]
```

Comment:
You restricted this one with the :create, but didn't restrict any of the other paths. I would have restricted all of them